kotlin/workflow-runtime/src/main/java/com/squareup/workflow/diagnostic/DebugSnapshotRecordingListener.kt

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.diagnostic
+
+import com.squareup.workflow.VeryExperimentalWorkflow
+import com.squareup.workflow.WorkflowAction
+import com.squareup.workflow.diagnostic.WorkflowHierarchyDebugSnapshot.ChildWorker
+import com.squareup.workflow.diagnostic.WorkflowHierarchyDebugSnapshot.ChildWorkflow
+import com.squareup.workflow.diagnostic.WorkflowUpdateDebugInfo.Kind.Passthrough
+import com.squareup.workflow.diagnostic.WorkflowUpdateDebugInfo.Kind.Updated
+import com.squareup.workflow.diagnostic.WorkflowUpdateDebugInfo.Source.Sink
+import com.squareup.workflow.diagnostic.WorkflowUpdateDebugInfo.Source.Subtree
+import com.squareup.workflow.diagnostic.WorkflowUpdateDebugInfo.Source.Worker
+
+/**
+ * A [WorkflowDiagnosticListener] that records [WorkflowHierarchyDebugSnapshot]s and
+ * [WorkflowUpdateDebugInfo]s and sends them to [debugger] after each render pass.
+ */
+@VeryExperimentalWorkflow
+@Suppress("TooManyFunctions")
```

Comment:
Should we change the threshold for `TooManyFunctions`, or just drop that check?

Reply:
I would rather keep it opt-out for now. It would be nice if there were some way for this suppression to propagate to subclasses but I don't think that's possible.